src/pivot-table/hooks/use-head-cell-dim.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useMemo, useState } from "react";
+
+interface UseHeadCellDim {
+  open: boolean;
+}
+
+export const useHeadCellDim = ({ open }: UseHeadCellDim) => {
+  const [isHovered, setIsHovered] = useState(false);
+
+  useEffect(() => {
+    if (!open) setIsHovered(false);
+  }, [open]);
+
+  const shadeOpacity = useMemo(() => {
+    let opacity = 0;
+    if (isHovered) opacity = 0.03;
```

Comment:
why not do this with css?

Reply:
See my comment about "The hovering color...", I don't there is way to achieve that with CSS. If there is that would be great tho.

Reply:
I think i've accomplished what we want here with [sass color functions](https://sass-lang.com/documentation/modules/color/) + [sass flow control rules](https://sass-lang.com/documentation/at-rules/control/if/) before, where it just automatically could detect for example text color based on background color. 

But the only reason I choose to do it like so is because in PVT, we are using inline styling which makes it hard for us to detect the element states like `:hover`, `:active` and ... in pure jss style

Reply:
I see. You can do it using styled-components. Have we decided anything there what we should do in this repo? Since a lot of things use styled components in sn-table, but not in pivot table

Reply:
The main concern with styled components is performance when scrolling. Since this particular component is rather static in nature, it should be "safe" to use styled components here. Which can solve the "hover" part of the feature